File: CosNet.API/Entities/Cosplay.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CosNet.API.Entities
{
    public class Cosplay
    {
        [Key]
        public int Id { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid CosplayId { get; set; }

        public Guid UserId { get; set; }

        [Required]
        public string Name { get; set; }

        public string Serie { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime DueDate { get; set; }
        public decimal Budget { get; set; }
        public string Status { get; set; }
    }
}

File: CosNet.AppUI.Android/app/src/main/java/cosnet/android/Entities/Cosplay.java
package cosnet.android.Entities;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;
import java.util.UUID;

@Entity
public class Cosplay implements Serializable {
   @PrimaryKey(autoGenerate = true)
   public int id;

   @ColumnInfo(name = "cosplay_id")
   @NonNull public String cosplayId;

   @ColumnInfo(name = "cosplay_name")
   @NonNull public String cosplayName;

   @ColumnInfo(name = "cosplay_series")
   @Nullable public String cosplaySeries;

   @ColumnInfo(name = "start_date")
   @Nullable public String startDate;

   @ColumnInfo(name = "due_date")
   @Nullable public String dueDate;

   @Nullable public Double budget;
   @NonNull public String status;

   public Cosplay(){
     this.cosplayId = UUID.randomUUID().toString();
   }
}
